src/System.Private.ServiceModel/tools/PRService/sync.ashx

```diff
@@ -0,0 +1,227 @@
+<%@ WebHandler Language="C#" Class="PullRequestHandler" %>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Web;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+// This script assumes that the repo is set up as follows: 
+
+//[remote "origin"]
+//	url = https://github.com/dotnet/wcf
+//	fetch = +refs/heads/*:refs/remotes/origin/*
+//	fetch = +refs/pull/*/head:refs/remotes/origin/pr/*      <-- allows us to run git checkout pr/{id} 
+
+public class PullRequestHandler : IHttpHandler
+{
+    private const string _gitExecutablePath = @"C:\Program Files (x86)\Git\bin\git.exe";
+    private const string _gitRepoPath = @"N:\git\wcf-test";
+
+    private const int _gitSanityExecutionTimeoutMilliseconds = 1000;
+    private const int _gitExecutionTimeoutMilliseconds = 60000;
+
+    public void ProcessRequest(HttpContext context)
+    {
+        string sha = context.Request.QueryString["sha"];
+        string pr = context.Request.QueryString["pr"];
+
+        context.Response.Write(string.Format("SHA = {0}<br/>", sha ?? "Unspecified"));
+        context.Response.Write(string.Format("PR = {0}<br/>", pr ?? "Unspecified"));
+
+        StringBuilder result = new StringBuilder();
+
+        bool success = CheckPrerequisites(result);
+
+        if (!success)
+        {
+            context.Response.Write(result);
+            return;
+        }
+
+        // Check inputs to the script
+        if (!string.IsNullOrEmpty(pr) && VerifyPrId(pr))
```

Comment:
NIT: if you have string.IsNullOrEmpty check in VerifyPrId, you can avoid it here.